code/modules/industrial_lift/tram_override_objects.dm

```diff
@@ -24,17 +24,66 @@
 	opacity = FALSE
 
 /obj/machinery/door/window/left/tram
+	icon = 'icons/obj/doors/tramdoor.dmi'
+	var/associated_lift = MAIN_STATION_TRAM
+	var/datum/weakref/tram_ref
+	name = "tram door"
+	desc = "Probably won't crush you if you try to rush them as they close. But we know you live on that danger, try and beat the tram!"
+
+/obj/machinery/door/window/left/tram/proc/find_tram()
+	for(var/datum/lift_master/lift as anything in GLOB.active_lifts_by_type[TRAM_LIFT_ID])
+		if(lift.specific_lift_id == associated_lift)
+			tram_ref = WEAKREF(lift)
+
 /obj/machinery/door/window/right/tram
+	icon = 'icons/obj/doors/tramdoor.dmi'
+	var/associated_lift = MAIN_STATION_TRAM
+	var/datum/weakref/tram_ref
+	name = "tram door"
+	desc = "Probably won't crush you if you try to rush them as they close. But we know you live on that danger, try and beat the tram!"
+
+/obj/machinery/door/window/right/tram/proc/find_tram()
+	for(var/datum/lift_master/lift as anything in GLOB.active_lifts_by_type[TRAM_LIFT_ID])
+		if(lift.specific_lift_id == associated_lift)
+			tram_ref = WEAKREF(lift)
 
 /obj/machinery/door/window/left/tram/Initialize(mapload, set_dir, unres_sides)
 	. = ..()
 	RemoveElement(/datum/element/atmos_sensitive, mapload)
 	associated_lift = MAIN_STATION_TRAM
+	INVOKE_ASYNC(src, PROC_REF(open))
+	find_tram()
 
 /obj/machinery/door/window/right/tram/Initialize(mapload, set_dir, unres_sides)
 	. = ..()
 	RemoveElement(/datum/element/atmos_sensitive, mapload)
 	associated_lift = MAIN_STATION_TRAM
+	INVOKE_ASYNC(src, PROC_REF(open))
+	find_tram()
+
+/obj/machinery/door/window/left/tram/open_and_close()
+	var/datum/lift_master/tram/tram_part = tram_ref?.resolve()
+	if(!open())
+		return
+	if(tram_part.travelling) //making a daring exit midtravel? make sure the doors don't go in the wrong state on arrival.
+		say("Emergency exit activated!")
+		return PROCESS_KILL
+	autoclose = TRUE
+	sleep(1 SECONDS)
+	if(!density && autoclose) //check for changed state
+		close()
+
+/obj/machinery/door/window/right/tram/open_and_close()
+	var/datum/lift_master/tram/tram_part = tram_ref?.resolve()
+	if(!open())
+		return
+	if(tram_part.travelling) //making a daring exit midtravel? make sure the doors don't go in the wrong state on arrival.
+		say("Emergency exit activated!")
+		return PROCESS_KILL
+	autoclose = TRUE
+	sleep(1 SECONDS)
+	if(!density && autoclose) //check for changed state
```

Comment:
I don't like how much we're copying and pasting on `left/tram` and `right/tram`. Is the direction the smartest thing to subtype off of here? Is there any reason this should not be `tram/left` and `tram/right`? If so, we should make this behavior an element and apply it to both.

Reply:
Yeah, I was thinking that, but this is how the windoors are set for the tram, and making them all a subtype makes them face the wrong way. So I guess there was a reason to do it that way.

Reply:
Surely we can just change the direction in the /left and /right subtypes of tram?

Reply:
tram is a subtype of left and right lol, backwards from what you'd expect

Reply:
That's what I mean, that should be fixed instead of adding more copy-paste between these

Reply:
Might as well fix that as part of this.

Reply:
Done, removed all the original tram code duplication.